sympy/solvers/ode/tests/test_single.py

```diff
@@ -1284,6 +1289,48 @@ def _get_examples_ode_sol_separable():
     }
 
 
+def _get_examples_ode_sol_1st_exact():
+    # Type: Exact differential equation, p(x,f) + q(x,f)*f' == 0,
+    # where dp/df == dq/dx
+    return {
+            'hint': "1st_exact",
+            'func': f(x),
+            'examples':{
+    '1st_exact_01': {
+        'eq': sin(x)*cos(f(x)) + cos(x)*sin(f(x))*f(x).diff(x),
+        'sol': [Eq(f(x), -acos(C1/cos(x)) + 2*pi), Eq(f(x), acos(C1/cos(x)))],
+        'slow': True,
+    },
+
+    '1st_exact_02': {
+        'eq': (2*x*f(x) + 1)/f(x) + (f(x) - x)/f(x)**2*f(x).diff(x),
+        'sol': [Eq(f(x), exp(C1 - x**2 + LambertW(-x*exp(-C1 + x**2))))],
+        'slow': True,
```

Comment:
This solution fails when tested in `checkodesol` and when we keep `sol=Eq(log(f(x)) + x/f(x) + x**2, C1)` it passes but test will still fail as dsolve output doesn't match with this.

Reply:
Is there a way to note that checkodesol is xfail for this test?

Reply:
No, currently xfail only stores the hints which it fails for.

Reply:
should there be way for getting xfail on checkodesol?

Reply:
Yes I think so.